### PR TITLE
Improved interface code generation in `slither.utils.code_generation`

### PR DESCRIPTION
**Pull request by webthethird**
Makes unrolling structs in function parameters and returns optional, and adds more robust testing

### CODE REVIEW

File: slither/utils/code_generation.py [hunk 1]
```diff
@@ -56,34 +116,63 @@ def generate_interface_function_signature(func: "FunctionContract") -> Optional[
...
+        if isinstance(param.type, ArrayType)
+        and isinstance(param.type.type, (UserDefinedType, ElementaryType))
+        else f"{str(param.type.type)} memory"
+        if isinstance(param.type, UserDefinedType)
+        and isinstance(param.type.type, (Structure, Enum))
+        else "address"
+        if isinstance(param.type, UserDefinedType) and isinstance(param.type.type, Contract)
+        else f"{param.type} {param.location}"
+        if param.type.is_dynamic
+        else str(param.type)
```
Comment (0xalpharush) [inline slither/utils/code_generation.py:164]: Could you extract this into a helper function please? 
Like 
```
def generate_interface_function_signature
     def helper
```

Comment (webthethird) [reply slither/utils/code_generation.py]: Done!

**Comment by webthethird** [comment]
Any other changes needed to get this one ready to merge?

**Review by 0xalpharush — APPROVED** [review]

**Comment by 0xalpharush** [comment]
It'd be cool to expose this as a tool for etherscan `slither-interface 0x0`

**Comment by webthethird** [comment]
@montyly I think this PR is ready to go!

**Pull request merged**